src/main/java/org/zalando/nakadi/exceptions/IllegalScopeException.java

```diff
@@ -0,0 +1,18 @@
+package org.zalando.nakadi.exceptions;
+
+import java.util.Set;
+
+public class IllegalScopeException extends RuntimeException {
```

Comment:
As for me I don't like idea with exception at all. 
I think it's better to return Result<T> and check if it's successful or not.
But I think it's another topic for our guidelines.